sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/text/TextTableProvider.java

```diff
@@ -93,6 +100,10 @@ public BeamSqlTable buildBeamSqlTable(Table table) {
                     : CSVFormat.DEFAULT);
         return new TextTable(
             schema, filePattern, new CsvToRow(schema, csvFormat), new RowToCsv(csvFormat));
+      case "json":
+        ObjectMapper objectMapper = getObjectMapper(schema);
```

Comment:
you can move this line to inside `JsonToRow` and `RowToJson` by only passing `schema` and construct what you need there.